### PR TITLE
Load security from OpenAPI v3

### PR DESCRIPTION
**Pull request by nguerrera**

### CODE REVIEW

File: adl/core/serialization/openapi/v3/security-schemes.ts [hunk 1]
```diff
@@ -1,7 +1,168 @@
...
+  }
+
+  // handle actual items next
+  for (const { key, value: securityScheme } of values(securitySchemes)) {
+    await $.process(processSecurityScheme, securityScheme);
+  }
+
+  // handle references last 
+  for (const { key, value: reference } of values(references)) {
+    // REVIEW
```
Comment (nguerrera) [inline adl/core/serialization/openapi/v3/security-schemes.ts:24]: I don't yet grasp how this part works and I'm not sure how to test this.

File: adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml [hunk 2]
```diff
@@ -1635,7 +1635,8 @@ components:
       name: Authorization
       in: header
       description: Amazon Signature authorization v4
-      x-amazon-apigateway-authtype: awsSigv4
+      #REVIEW
+      #x-amazon-apigateway-authtype: awsSigv4
```
Comment (nguerrera) [inline adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml:1639]: I wasn't sure how this is supposed to be handled either

Comment (fearthecowboy) [reply adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml:1639]: extensions that we don't recognize, we're just going to add to the element's attic. I have a function to do that -- look at `core/openapi/v3/info.ts`  --  `await addExtensionsToAttic(result, contact);`

File: adl/core/serialization/openapi/v3/security-schemes.ts (hunk 1 above)
Comment (fearthecowboy) [reply adl/core/serialization/openapi/v3/security-schemes.ts:24]: I just finished refactoring this to remove the stupid. 

my next PR makes this easy.

File: adl/openapi/v3/security.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ export enum SecurityType {
   ApiKey = 'apiKey',
   Http = 'http',
   OAuth2 = 'oauth2',
-  OpenIDConnect = 'openIdConnect'
+  OpenIdConnect = 'openIdConnect'
```
Comment (fearthecowboy) [inline adl/openapi/v3/security.ts:21]: 👍

File: adl/core/serialization/openapi/v3/security-schemes.ts (hunk 1 above)
Comment (fearthecowboy) [reply adl/core/serialization/openapi/v3/security-schemes.ts:24]: And more to the point, security schemes doesn't support references.
so, just remove that part

File: adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml (hunk 2 above)
Comment (nguerrera) [reply adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml:1639]: should we move that somewhere else or export it from info?

Comment (fearthecowboy) [reply adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml:1639]: I have already moved it in my work tree. go ahead and just export it and I'll clean up when I merge.

File: adl/core/serialization/openapi/v3/security-schemes.ts (hunk 1 above)
Comment (nguerrera) [reply adl/core/serialization/openapi/v3/security-schemes.ts]: Done.

File: adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml (hunk 2 above)
Comment (nguerrera) [reply adl/core/test/scenarios/v3/single/input/aws-app-insights.yaml]: Done.

**Pull request merged**